lib/wasi/src/runtime/module_cache/shared.rs

```diff
@@ -0,0 +1,42 @@
+use std::{collections::HashMap, sync::Arc};
+
+use tokio::sync::RwLock;
+use wasmer::{Engine, Module};
+
+use crate::runtime::module_cache::{CacheError, CompiledModuleCache};
+
+/// A [`CompiledModuleCache`] based on a
+/// <code>[Arc]<[RwLock]<[HashMap]<[String], [Module]>>></code> that can be
+/// shared.
+#[derive(Debug, Default, Clone)]
+pub struct SharedCache {
+    modules: Arc<RwLock<HashMap<String, Module>>>,
```

Comment:
We can use `DashMap` instead, it will be more performant
https://crates.io/crates/dashmap/2.1.0

Reply:
Ooh, good thinking!